include/singa/io/encoder.h

```diff
@@ -30,13 +31,15 @@ class Encoder {
   public:
     Encoder() { }
     virtual ~Encoder() { }
+    
+    virtual void Setup(const LayerConf& conf) = 0;
```

Comment:
suppose some parameters are set in LayerConf
currently not sure about what params we will support for users to config